ampcrowd/basecrowd/models.py

```diff
@@ -150,3 +171,43 @@ def add_model_rels(self):
         # responses pertain to a task
         self.add_rel(self.response_model, self.task_model, models.ForeignKey,
                      'task', 'responses')
+
+
+class TemplateResource(models.Model):
+    name = models.CharField(max_length=200)
+    content = models.TextField()
+    direct_dependencies = models.ManyToManyField('self', symmetrical=False, related_name="upstream_dependencies")
+    direct_requirements = models.ManyToManyField('self', symmetrical=False, related_name="upstream_requirements")
```

Comment:
Ok, then `imports` and `requirements` sound like better names.

But I'm wondering if it wouldn't be cleaner to place these in separate models, one model for `TemplateResource` (TR) objects that represent interface HTML templates, and one model for `RenderingResource` (RR) objects that represent an importable `script` tag. Then TRs could have one field pointing to required TRs, and one field pointing to required RRs. The RRs for a hierarchy of TRs could be marshaled easily with code like `RR.objects.filter(tr__in=[list of TRs being rendered])`